crypto/evp/mac_lib.c

```diff
@@ -189,3 +189,65 @@ int EVP_MAC_names_do_all(const EVP_MAC *mac,
 
     return 1;
 }
+
+unsigned char *EVP_Q_mac(OSSL_LIB_CTX *libctx, const char *name, const char *propq,
+                         const char *subalg, const OSSL_PARAM *params,
```

Comment:
Just an annoying thought: could we varargs this function and have the _subalg_ as the first extra argument?  Not all MACs need this value and those that do treat it as different algorithms.

Reply:
I don't think it's worth doing that if it's just about an optional parameter for which the given type fits all uses where the parameter is needed.
Do you have a better name than `subalg` that we could use?

Reply:
BTW, in another library I introduced the label `OPTIONAL` defined as empty/no-op:
```
#define OPTIONAL
```
This provides a simple way of marking optional (pointer-type) function parameters.
Using it, the prototype would become
```
unsigned char *EVP_Q_mac(OPTIONAL OSSL_LIB_CTX *libctx, const char *name, 
                         OPTIONAL const char *propq,
                         OPTIONAL const char *subalg, OPTIONAL const OSSL_PARAM *params,
                         const void *key, size_t keylen,
                         const unsigned char *data, size_t datalen,
                         OPTIONAL unsigned char *out, size_t outsize, 
                         OPTIONAL unsigned int *outlen)
```